copyerror.go

```diff
@@ -0,0 +1,61 @@
+/*
+Copyright The ORAS Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package oras
+
+import "fmt"
+
+// CopyErrorOrigin defines the source of a copy error.
+type CopyErrorOrigin string
+
+const (
+	// CopyErrorOriginSource indicates the error occurred at the source side.
+	CopyErrorOriginSource CopyErrorOrigin = "source"
+
+	// CopyErrorOriginDestination indicates the error occurred at the destination side.
+	CopyErrorOriginDestination CopyErrorOrigin = "destination"
+
+	// CopyErrorOriginInternal indicates the error occurred internally.
+	CopyErrorOriginInternal CopyErrorOrigin = "internal"
+)
+
+// CopyError represents an error encountered during a copy operation.
+type CopyError struct {
+	Op     string
```

Comment:
It looks like most `Op` values are not very meaningful to end users 😂

Reply:
I refactored the code a little bit, and now it should be able to help users to get a basic idea on where the error occurred.